akka-http-core/src/main/scala/akka/http/impl/util/StreamUtils.scala

```diff
@@ -59,7 +59,7 @@ private[http] object StreamUtils {
         override def onPull(): Unit = pull(in)
 
         override def onUpstreamFailure(ex: Throwable): Unit = {
-          promise.failure(ex)
+          promise.tryFailure(ex)
```

Comment:
If I don't do this the futures fail with double failing the same promise instead of the double-materialization error - seems valid IMO, but can someone confirm my thinking this change should be harmless?

// cc @drewhk @jrudolph @2beaucoup